### ISSUE TITLE
Add transpilation step to use EsNext features

### ISSUE THREAD

**Issue by fr-ser** [react-bootstrap/react-bootstrap#4599]
## Describe the solution you'd like

I would like to specify supported browsers and let babel do the work: https://babeljs.io/docs/en/babel-preset-env#browserslist-integration

## Describe alternatives you've considered

Current one: Write code with browser features in mind

## Additional context

An example would be using features like Array.includes() instead of Array.indexOf() !== -1.
This reads betters and avoids confusion for new developers, who are not aware of our transpilation / browser support limitations.

Is this something you would consider useful for this repo?

**Comment by taion** [comment]
yup, we should use corejs 3 with the runtime transform in https://github.com/react-bootstrap/configs/blob/master/packages/babel-preset/index.js

**Comment by taion** [comment]
which will transpile instance methods for us. currently we're on corejs 2 which only does statics

**Comment by fr-ser** [comment]
I would be happy to take this, but I don't fully get the project structure right now.

I assumed I would have to adjust the [build file in this repo](https://github.com/react-bootstrap/react-bootstrap/blob/master/tools/build.js)

Is there some documentation on how the configs repo interacts with this one? Or are there some instructions about setting up the build environment locally (or at least replicate it)?

**Comment by taion** [comment]
I think it would be something like... clone `react-bootstrap/configs`, edit the relevant individual package, and use `yarn link` to link it into this current repo, as the Babel config here references the Babel preset there.

Though it's sort of an open question on whether we want to pull in core-js in the runtime transform. It's nice to be able to e.g. use `includes`, but in our experience the runtime transform has been pretty indiscriminate about pulling in unnecessary polyfills as well.

cc @jquense thoughts?
